src/components/shared/react-hook-form-inputs/RHFTextField.tsx

```diff
@@ -36,27 +37,34 @@ export const RHFTextField: React.FC<RHFTextFieldProps> = ({
   rules,
   size = 'small',
   rows,
+  indexFieldArray,
   ...props
 }) => {
   const { formState } = useFormContext()
   const { t } = useTranslation()
 
-  const error = formState.errors[name]?.message as string | undefined
+  const error =
+    indexFieldArray !== undefined
+      ? //@ts-ignore
+        (formState.errors[name]?.[indexFieldArray]?.message as string | undefined)
+      : (formState.errors[name]?.message as string | undefined)
 
-  const { accessibilityProps, ids } = getAriaAccessibilityInputProps(name, {
+  const fieldName = indexFieldArray !== undefined ? `${name}.${indexFieldArray}` : name
+
+  const { accessibilityProps, ids } = getAriaAccessibilityInputProps(fieldName, {
```

Comment:
I was already forgetting why this is made like this, could you please add a comment?